networkx/algorithms/chordal.py

```diff
@@ -344,3 +347,79 @@ def _connected_chordal_graph_cliques(G):
                 raise nx.NetworkXError("Input graph is not chordal.")
         cliques.add(frozenset(clique_wanna_be))
         return cliques
+
+
+@not_implemented_for('directed')
+def complete_to_chordal_graph(G):
+    r"""Complete to chordal graph
+
+        Enhances the graph by edges to receive a chordal graph.
+
+        Parameters
+        ----------
+        G : NetworkX graph
+            Undirected graph
+
+        Returns
+        -------
+        H : NetworkX graph
+            The chordal enhancement of G
+        alpha : Dictionary
+                The elimination ordering of nodes of G
+
+        Notes
+        ------
+        A graph G=(V,E) is called chordal if for each cycle with
+        length bigger than 3, two non adjacent nodes are connected
+        by an edge (called chord). There are different approaches
+        to calculate the chordal enhancement of a graph.
+        The algorithm used here is called MCS-M and gives a minimal
+        triangulation of graph. Note that this triangulation must not
+        be minimum.
+
+        https://en.wikipedia.org/wiki/Chordal_graph
+
+        References
+        ----------
+        .. [1] Berry, Anne & Blair, Jean & Heggernes, Pinar & Peyton, Barry. (2004).
+               Maximum Cardinality Search for Computing Minimal Triangulations of Graphs.
+               Algorithmica. 39. 287-298. 10.1007/s00453-004-1084-3.
+
+        Examples
+        --------
+        >>> import networkx as nx
+        >>> from networkx.algorithms.chordal import complete_to_chordal_graph
+        >>> G = nx.wheel_graph(10)
+        >>> H,alpha = complete_to_chordal_graph(G)
+    """
+    if G is None:
+        raise ValueError("Expected NetworkX graph!")
+    H = G.copy()
+    alpha = {node: 0 for node in H.nodes()}
+    if nx.is_chordal(H):
+        return H, alpha
+    chords = []
+    weight = {node: 0 for node in H.nodes()}
+    unnumbered_nodes = list(H.nodes())
+    for i in range(len(H.nodes()), 0, -1):
+        # get the node in unnumbered_nodes with the maximum weight
+        z = [node for node in unnumbered_nodes
+             if weight.get(node) == max([weight.get(i)
+                                         for i in unnumbered_nodes])][0]
+        unnumbered_nodes.remove(z)
+        alpha[z] = i
+        update_nodes = []
+        sub_graph_unnumbered = nx.subgraph(H, unnumbered_nodes + [z])
+        for y in unnumbered_nodes:
+            # weights of nodes on a path between y and z should have smaller weight than y
+            lower_nodes = [node for node in unnumbered_nodes
```

Comment:
The condition is not enough. One have to check if there is a path between y and z, since later you also check for a path on a graph with special nodes. And I think often these graphs are connected and you do not have a forest. But moving the weight retrieval for y in the line before is useful.

Reply:
Wait...  if (z,y) is an edge then there is a path from y to z right? (directed graphs are not implemented)
If (z,y) is NOT an edge then we have create the subgraph and check for the path.

Other changes you should make:

- replace ```weight.get(stuff)``` with ```weight[stuff]``` throughout.
- replace ```nx.subgraph(H, stuff)``` with ```H.subgraph(stuff)```

Reply:
Done the changes.

Reply:
I still have a question about avoiding the search for a path in a subgraph.
As written, even if y and z are connected by an edge, you still create a (potentially large) subgraph and search for a path between y and z in that subgraph. But you KNOW that there is a path between y and z in that subgraph because an edge exists in H and the subgraph is induced on y, z and other nodes. So the edge is a path between y and z.

Can we avoid the subgraph creation by code like:

    if G.has_edge(z,y):
        update_nodes.append(y)
    else:
        ywt = weight[y]
        lower_nodes = [node for node in unnumbered_nodes if weight[node] < ywt]
        if nx.has_path(H.subgraph(lower_nodes + [z, y]), z, y):
            update_nodes.append(y)
            chords.add((z, y))   # make chords a set earlier...